CRM/Core/Form.php

```diff
@@ -2725,7 +2727,7 @@ protected function getContactIDIfAccessingOwnRecord() {
     if (!$contactID) {
       return FALSE;
     }
-    if ($contactID === $this->getLoggedInUserContactID()) {
+    if ($contactID === CRM_Core_Session::getLoggedInContactID()) {
```

Comment:
We should probaly cast $contactID to an int here to be sure we are strict on both sides

Reply:
@eileenmcnaughton at L2726 above there is int cast on $contactID assignment. So I think the conditions obeys strict checking

Reply:
@eileenmcnaughton Yes, this one is fine because we are casting both to int before doing the check.